.github/workflows/ci.yml

```diff
@@ -269,23 +319,47 @@ jobs:
   system-tests-linux:
     runs-on: ubuntu-latest
     needs: build
+    env:
+      # Disable telemetry, because we use many small projects
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      REQNROLL_TELEMETRY_ENABLED: 0
 
     steps:
     - uses: actions/checkout@v4
+    - name: Set Runner ImageVersion to Env
+      shell: pwsh
+      run: |
+        Write-Output "ImageVersion=$env:ImageVersion" >> $env:GITHUB_ENV
+    - name: Cache RRC folder
+      uses: actions/cache@v4
+      with:
+        path: /tmp/RRC
+        key: RRC-SystemTests-${{ runner.os }}-${{ env.ImageVersion }} # Use the runner version as key, so if new dependencies get installed we get a new cache entry
+    - name: Download packages
+      uses: actions/download-artifact@v4
+      with:
+        name: packages-v${{ needs.build.outputs.product_full_version }}
+        path: GeneratedNuGetPackages/${{ needs.build.outputs.product_configuration }}/
+    - name: Download SystemTests
+      uses: actions/download-artifact@v4
+      with:
+        name: SystemTests
+        path: Tests/Reqnroll.SystemTests/bin/
+    - name: Setup dotnet
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: | 
+          6.0.x
+          7.0.x
+          8.0.x
+          9.0.x
```

Comment:
Not for this PR, but would there be value in using a container image that already had these dependencies installed?

Reply:
Forgive me for not being clear: I meant using a container as the runner for the job https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container

This would mean you could pre-install the required runtimes as part of the container and avoid any installation during the execution of the job; the container would already contain everything for us.